packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -278,9 +292,14 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
       mounted.current = true;
       return;
     }
-    const value = options[selectedIndex]?.value;
-    onChangeProp.current?.(value);
-  }, [options, selectedIndex]);
+    const currentValue = optionsRef.current[selectedIndex]?.value;
+
+    // If current value is the same as selected value, do not call user defined onChange
+    if (currentValue === valuePropRef.current || selectedIndex === -1) {
+      return;
+    }
```

Comment:
I think this check is also not needed now that we don't have `options` or `value` in the deps array

Reply:
This check is very much needed because user defined onChange is being called two times:
1. User programmatically changes valueProp and useEffect with [options, valueProp] changes selectedIndex. 
2. This useEffect is called when selectedIndex change and calls user defined onChange.

Reply:
But you removed this check in the previous PR, does it not reintroduce that bug?

Reply:
Previous check used not "ref'ed" value of valueProp. That's why it caused infinite loop. Now it should work as expected.
I probably missed this double onChange call in previous PR, so fixing my own bug.